enhancements/compliance-operator.md

```diff
@@ -0,0 +1,246 @@
+   to create the MachineConfigs or other resources that need to be created.
+   What would be a safe scheme to do this so that we minimize the risks in case
+   the operator is compromised?
+
+## Design Details
+
+The `compliance-operator` would use the following API to track a compliance scan status:
+
+### API Specification
+The proposed compliance scan API instance with all the properties set:
+
+```
+apiVersion: complianceoperator.compliance.openshift.io/v1alpha1
+kind: ComplianceScan
+metadata:
+  name: example-scan
+spec:
+  contentImage: quay.io/compliance-as-code/openscap-ocp4
+  rule: xccdf_org.ssgproject.content_rule_no_empty_passwords
+  profile: xccdf_org.ssgproject.content_profile_ospp
+  content: ssg-ocp4-ds.xml
+```
+
+A minimal `ComplianceScan` instance with only the required properties set:
+```
+apiVersion: complianceoperator.compliance.openshift.io/v1alpha1
+kind: ComplianceScan
+metadata:
+  name: example-scan
+spec:
+  profile: xccdf_org.ssgproject.content_profile_ospp
+  content: ssg-ocp4-ds.xml
+```
+
+The content would be provided with an image separate from the scanner
+image. This way, we can allow cluster administrator to provide their
+own content easily. On the other hand, the `contentImage` property is not
+required to be set explicitly and would default to an image with officially
+supported content.
+
+## Need Feedback
+
+How we present the remediations and allow the cluster administrator to execute
+them is something that should be discussed more.
+
+On a high level, there will be three kinds of remediations:
+ * General free-form text guidances. For example, your IDP must support 2FA.
+ * More specific advise, but still needs to be applied manually. For example,
+   "configure a message of the day so that a legal notice gets displayed after
+   login by creating a ConfigMap called `motd` in the `openshift` namespace."
+ * Gaps that can be remediated in a completely autonomous fashion. For example
+   making sure that the `auditd` service is enabled.
+
+At first, we would like to display the remediation in the HTML report so that
+the administrator can copy the advise and, if it's possible to apply directly,
+do that.
+
+However, when it comes to applying the remediation, there are two options:
+
+* Output the remediations to yaml files that the user can download and apply.
+  The caveat of this is that it makes it cumbersome: First check the report,
+  then download the remediations, then apply them manually.
+
+* Add the capability so that the opreator can schedule a workload that will
+  apply these remediations. The workflow would be as follows: Check the report
+  then modify the CR so that the remediation is applied. This is more
+  automatic and user-friendly, however, it requires us giving a lot of
+  privileges to the workload, which might not be ideal.
+
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
```

Comment:
How about:

A CI test that
- Rolls out the operator
- Ensures the operator is running
- Executes a scan
- Ensures a result is produced and the scheduled container is cleaned up

Reply:
Sounds good, I worded the paragraph a little differently, but I think it's still quite close to what you suggested.